src/printer.js

```diff
@@ -2151,63 +2155,78 @@ function printJSXChildren(path, options, print) {
 function printJSXElement(path, options, print) {
   const n = path.getValue();
 
-  // turn <div></div> into <div />
+  // Turn <div></div> into <div />
   if (isEmptyJSXElement(n)) {
     n.openingElement.selfClosing = true;
     delete n.closingElement;
   }
 
-  // if no children, just print the opening element
+  // If no children, just print the opening element
   const openingLines = path.call(print, "openingElement");
   if (n.openingElement.selfClosing) {
     assert.ok(!n.closingElement);
     return openingLines;
   }
 
   const children = printJSXChildren(path, options, print);
-  let hasAnyHardLine = false;
+  let forcedBreak = false;
```

Comment:
I had to change this code because when a line breaked in JSX, it forced the entire JSX element to break (all children). I cleaned it up a little, but the main thing was to suppress break propagation (which you can do by using `conditionalGroup`).